src/memory.rs

```diff
@@ -157,63 +158,49 @@ pub struct ProcessMemoryStats {
 }
 
 impl ProcessMemoryStats {
-    pub fn new() -> Self {
-        Self::default()
-    }
-
     /// Update the process memory stats
     /// # Examples
+    ///
+    /// ```no_run
+    /// let mut pms = zmem::ProcessMemoryStats::default();
+    /// pms.update(1, "cmd".into()).unwrap();
     /// ```
-    /// let mut pms = ProcessMemoryStats::new();
-    /// pms.update(1)?;
-    /// ```
-    pub fn update(&mut self, pid: &u32) -> Result<(), AnyError> {
-        self.command = get_cmd(*pid)?;
-        if self.command.len() > 50 {
-            self.command.truncate(50);
+    pub fn update(&mut self, pid: u32, mut cmd: String) -> Result {
+        if cmd.len() > 50 {
+            cmd.truncate(50);
         }
-
-        self.pid = *pid;
-
-        let smaps_file = File::open(format!("/proc/{}/smaps", pid))?;
-        let mut reader = BufReader::new(smaps_file);
-        let mut line = String::new();
-
-        let mut mem_values = (0, 0, 0, 0);
```

Comment:
This is here for data locality, you are reducing performance.

Reply:
Also, bufreader is a more performant way to read files.

Reply:
> bufreader is a more performant way to read files

It doesn't, `BufReader` is  only good when you have frequent small read.  

Also `fs::read_to_string` buffer the entire contents in memory.